src/Network/Pusher/Data.hs

```diff
@@ -1,4 +1,5 @@
 {-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE RecordWildCards #-}
```

Comment:
That's a good point about type safety. Since I'm finding that the majority of the fields in these notification payloads are optional, what would you think about a hybrid approach? Something along the lines of `data Notification = { alert :: ApnsAlert, interest :: Interest, options :: HS.HashSet ApnsProperties}`.

I think something like this end up being nice to work with:

```haskell
Notification 
  --Required parameters
  (AlertText "You've got mail!") 
  (Interest "foobar") 
  --Optional parameters
  (Set.fromList
    [ (WebhookUrl "http://requestb.in/1nxrsss1")
    , (WebhookLevel Debug)
    ])
```

as opposed to (potentially):

```haskell
Notification 
  --Required parameters
  (AlertText "You've got mail!") 
  (Interest "foobar") 
  --Optional parameters
  (Just $ WebhookUrl "http://requestb.in/1nxrsss1")
  (Just $ WebhookLevel Debug)
  Nothing
  Nothing
  Nothing
  (Just 1)
```